nixos/modules/services/development/lorri.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.lorri;
+  socketUnit = "lorri";
+  socketPath = "lorri/daemon.socket";
+in with lib; {
+  options = {
+    services.lorri = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          This option enables a systemd socket unit that listens on the
+          well-known address used by lorri, a nix-shell replacement for project
+          development. When a lorri client connects to the socket, systemd
+          starts the lorri daemon as a service.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.sockets.${socketUnit} = {
+      description = "Socket for Lorri Daemon";
+      enable = true;
```

Comment:
You don't need to explicitly enable this, similar to any other units, this is the [default](https://github.com/NixOS/nixpkgs/blob/2e52f98ddbe9ddd91f65f8df471d84e1e4077bdf/nixos/modules/system/boot/systemd-unit-options.nix#L32-L44).